toy-submission/main.py

```diff
@@ -0,0 +1,147 @@
+from typing import Union, List, Dict
+from pathlib import Path
+from fastapi import FastAPI
+from pydantic import BaseModel
+import logging
+from dataclasses import dataclass
+
+
+# Lit-llama imports
+from helper import neurips_generate
+import sys
+import time
+import warnings
+from pathlib import Path
+from typing import Optional
+
+import lightning as L
+import torch
+
+torch.set_float32_matmul_precision("high")
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_llama import LLaMA, Tokenizer
+from lit_llama.utils import lazy_load, llama_model_lookup, quantization
+
+app = FastAPI()
+
+logger = logging.getLogger(__name__)
+# Configure the logging module
+logging.basicConfig(level=logging.INFO)
+
+quantize = "llm.int8"
+
+checkpoint_path = Path("checkpoints/lit-llama/7B/lit-llama.pth")
+tokenizer_path: Path = Path("checkpoints/lit-llama/tokenizer.model")
+
+precision = "bf16-true"
+fabric = L.Fabric(devices=1, precision=precision)
+
+logger.info("Loading model ...")
+t0 = time.time()
+with lazy_load(checkpoint_path) as checkpoint:
+    name = llama_model_lookup(checkpoint)
+
+    with fabric.init_module(empty_init=True), quantization(mode=quantize):
+        model = LLaMA.from_name(name)
+
+    model.load_state_dict(checkpoint)
+logger.info(f"Time to load model: {time.time() - t0:.02f} seconds.")
+
+model.eval()
+model = fabric.setup(model)
+
+tokenizer = Tokenizer(tokenizer_path)
+
+
+class ProcessRequest(BaseModel):
+    prompt: str
+    num_samples: int = 1
+    max_new_tokens: int = 50
+    top_k: int = 200
```

Comment:
do we need top_k, temperature and seed? As in how do we know for sure if people did use them in their code. I guess if they dont and their results are not reproducible then shucks too bad?